controllers/factory/scrapes.go

```diff
@@ -798,11 +803,11 @@ func CreateVMServiceScrapeFromService(ctx context.Context, rclient client.Client
 			Labels:          service.Labels,
 			Annotations:     service.Annotations,
 		},
-		Spec: victoriametricsv1beta1.VMServiceScrapeSpec{
-			Selector:  metav1.LabelSelector{MatchLabels: service.Spec.Selector},
-			Endpoints: endPoints,
-		},
+		Spec: *serviceScrapeSpec,
 	}
+	scrapeSvc.Spec.Selector = metav1.LabelSelector{MatchLabels: service.Spec.Selector}
+	scrapeSvc.Spec.Endpoints = append(endPoints, scrapeSvc.Spec.Endpoints...)
```

Comment:
WDYT about patching generated endpoints?

Reply:
Could you please elaborate more on this? How can I patch generated endpoints?

Reply:
Could you please elaborate more on this? @f41gh7 

I assume that you want to patch any endpoints generated by the operator. Since endpoints are array, not map, which isn't straightforward to merge, could you please elaborate how should 2 sets of endpoints be merged?

Reply:
I'll try to add it as follow-up.

I think, it's possible to match array item by `endpoint.Port` and override generated values (port and path). Usually generated endPoints contains only 1 item, so it should to be easy to match and update. It must cover the case, when you want to add some settings for application endpoint.